### DISCUSSION TITLE
Representing integer enums

### DISCUSSION THREAD

**Discussion by JeongJuhyeon** [prisma/prisma#12402] [Q&A]
Let's say I have a table _payment_. It has an integer column _status_. It can be represented by the typescript enum 

```typescript
enum PaymentStatus {
  CREATED = 100,
  ONGOING = 120,
  COMPLETE = 200,
  CANCELLED = 300
}
```

Is there any way to connect this to Prisma? Else you just end up getting

```typescript
const payment = prismaService.payments.findOne(...);
payment.status // number
```

This is rather uncomfortable as it lacks type safety on create/update.
Plus it means you can't make use of type narrowing: 
```typescript
  if (s === PaymentStatus.ONGOING || s === PaymentStatus.COMPLETE) {

  } else {
    s // s: PaymentStatus.CREATED | PaymentStatus.CANCELLED
  }
 ```

**Comment by nurul3101** [comment]
Hey @JeongJuhyeon 👋 

We have a similar [Feature Request](https://github.com/prisma/prisma/issues/273) to allow custom values to be mapped to enum.
If you could leave a 👍 to the request and mention your use case it would allow our engineering and product team to prioritise it.

**Reply by JeongJuhyeon** [reply]
Done.

**Comment by dipanshSaxenaVivo** [comment]
### Support for Numeric Enum Values and Bitwise Operations

Currently, Prisma only supports string-based enums. However, many use cases require enums with numeric values, especially when working with bitwise operations.
I propose adding support for numeric enum values in Prisma. This would allow developers to define enums like this:

`
enum Role {
  USER = 1
  SUPER_ADMIN = 2
  SUPPORT = 4
  ADMIN = 8
}
`
With numeric enum values, bitwise operations would automatically work in Prisma, enabling use cases like role-based access control.

Benefits:
- Improved performance: Bitwise operations are faster than string-based comparisons.
- Simplified code: No need for additional libraries or workarounds to handle bitwise operations.
- Enhanced security: Role-based access control is a common use case that would benefit from numeric enum values.

Use cases:
- Role-based access control
- Permissions management
- Flag-based enumeration

Thanks for considering this feature request!

**Reply by UltraMarco** [reply]
I'd like to see this feature implemented. It would be very useful to me.
Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
